addons/trenches/functions/fnc_placeCamouflage.sqf

```diff
@@ -0,0 +1,79 @@
+#include "script_component.hpp"
+/*
+ * Author: chris579, Salbei
+ * Places camouflage on a trench
+ *
+ * Arguments:
+ * 0: trench <OBJECT>
+ * 1: unit <OBJECT>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [TrenchObj, Unit] call ace_trenches_fnc_placeCamouflage
+ *
+ * Public: No
+ */
+
+params [
+    ["_trench", objnull, [objnull]],
+    ["_unit", objnull, [objnull]]
+];
+
+if (isNull _trench || {count getArray (configFile >> "CfgWorlds" >> worldName >> "camouflageObjects") == 0}) exitWith {};
+
+private _fnc_onFinish = {
+    (_this select 0) params ["_unit", "_trench"];
+
+    private _camouflageObjects = getArray (configFile >> "CfgWorlds" >> worldName >> "camouflageObjects");
+    private _statusNumber = _trench getVariable [QGVAR(trenchCamouflageStatus), 0];
+    private _statusString = "";
+    _statusNumber = _statusNumber +1;
+
+    if (_statusNumber <10) then {
+      _statusString = ("0" + str(_statusNumber));
+   }else{
+      _statusString = str(_statusNumber);
+   };
+
+
+    private _placedObjects = [];
+    private _camouflageObjectsArray = _trench getVariable [QGVAR(camouflageObjects), []];
+
+    {
+        private _object = createSimpleObject [selectRandom _camouflageObjects, [0,0,0]];
+        _object attachTo [_trench, getArray(_x)];
+
+        if (is3DEN) then {
+            _object setVariable [QGVAR(positionData), getArray(_x),true];
+        };
+
+        _placedObjects pushBack _object;
+    } forEach (configProperties [configFile >> "CfgVehicles" >> (typeof _trench) >> ("CamouflagePositions" + _statusString)]);
+
+    reverse _camouflageObjectsArray;
+    _camouflageObjectsArray pushBack _placedObjects;
```

Comment:
Oh how much I hate this reversing just to do a `pushFront`..
Intercept CBA has a pushFront btw ;) :D 
But kinda dumb to copy this huge file just to make use of intercept in this small place

Reply:
Not making it Intercept compatible for such a smal thing